src/utilities/string.ts

```diff
@@ -8,3 +10,27 @@ export function toTitle(string: string) {
     .replace(/(?:^|\s|-)\S/g, (x) => x.toUpperCase())
     .replace(/-/g, ' ');
 }
+
+const replaceNonUrlSafeCharacters = (base64String: string) =>
+  base64String.replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/g, '');
+const restoreNonUrlSafeCharacters = (base64String: string) =>
+  base64String.replace(/-/g, '+').replace(/_/g, '/');
+
+export const UrlSafeBase64 = {
+  encode: (rawString: string) => {
+    if (isNode()) {
+      return replaceNonUrlSafeCharacters(Buffer.from(rawString, 'latin1').toString('base64'));
```

Comment:
nit: just personal preference avoid deep nested call stacks

```js
const base64Encoded = Buffer.from(rawString').toString('base64');
return replaceNonUrlSafeCharacters(base64Encoded);
```

Reply:
or one more option

```js
const base64Encoded = isNode() ?  Buffer.from(rawString').toString('base64') :  btoa(rawString);
return restoreNonUrlSafeCharacters(base64Encoded);
```

could be more straightforward?

Reply:
just curious why we are using `latin1` encode method instead of default `utf8`?

Reply:
> or one more option
> 
> ```js
> const base64Encoded = isNode() ?  Buffer.from(rawString').toString('base64') :  btoa(rawString);
> return restoreNonUrlSafeCharacters(base64Encoded);
> ```
> 
> could be more straightforward?

i prefer the latter.

Reply:
> just curious why we are using `latin1` encode method instead of default `utf8`?

`btoa` and `atob` use Latin1 encoding, so force `buffer` to use `Latin1` encoding to guarantee the behavior of `encode` and `decode` stay the same on both Node and
 web browser.